### PR TITLE
Add "state of the field" to the paper

### LINKED ISSUE #129: "State of the Field" section in JOSS paper
From the review in https://github.com/openjournals/joss-reviews/issues/5028:

>State of the field: Do the authors describe how this software compares to other commonly-used packages?

And reviewer comment:

>See the JOSS required item above to please add "State of the Field" to cover how this package compares to other packages. If none such exists (which I doubt it, since stochastic local search is well-known), please make the contribution explicit. If there are other tools, please cover mainstream ones, if any and mention where each would benefit the most (e.g., genetic algorithms for multi-point solutions vs. alns)

There's some references already written down in the review that might be useful. I'm thinking ten-or-so references at most (incl. some open source Python/other optimization packages), and text similar in size to the statement of need.

@leonlan can you have a go at this?

### PR DESCRIPTION
**Pull request by leonlan**
Closes #129:
- [x] Add references to other metaheuristics software libraries
- [x] Add state of field section about related software libraries.

### CODE REVIEW

**Comment by leonlan** [comment]
I wrote my version of the "the state of field" section.

It was a bit of a back-and-forth between adding some sentences to the "statement of need" section and a new "state of field" section, but I finally decided that an entire new section was better.

**Comment by N-Wouda** [comment]
I'll have a look at this later tonight!

File: paper/paper.md [hunk 1]
```diff
@@ -41,6 +41,17 @@ Our `alns` package, by contrast, offers a clear and problem-agnostic API for usi
...
+The most widely-used packages generally focus on population-based evolutionary algorithms and multi-objective optimization, for example `DEAP` [@Fortin2012-DEAPEvolutionaryAlgorithms], `ECJ` [@Scott2019-ECJ20General], `jMetal` [@Durillo2011-JMetalJavaFramework], `Metaheuristics.jl` [@Mejia-de-Dios2022-MetaheuristicsJuliaPackage], and `PySwarms` [@Miranda2018-PySwarmsResearchToolkit]. 
+These libraries provide limited functionality for single-trajectory algorithms, which is the class of algorithms that ALNS belongs to.
+Among libraries that focus on single-trajectory algorithms are `Paradiseo` [@Dreo2021-ParadiseoModularFramework], which is implemented in C++, and `Chips-n-Salsa` [@Cicirello2020-ChipsnSalsaJavaLibrary], which is implemented in Java.
+However, these libraries do not provide an implementation of the ALNS algorithm out of the box.
+Finally, the ALNS library by @Santini2019-AdaptiveLargeNeighbourhood is a problem-agnostic implementation of ALNS in C++.
+This library provides a large number of acceptance criteria, but only provides the roulette wheel mechanism as operator selection scheme.
```
Comment (N-Wouda) [inline paper/paper.md:53]: I think we can phrase this a bit clearer, but I'll have a look at that tomorrow.

**Review by N-Wouda — APPROVED** [review]
Let's get this merged in! Tomorrow I'll finish up the paper with a bit about what users need to provide/minimal example as well.

**Pull request merged**